tests/utils/Constraint/ImageHasSource.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace PerformanceLab\Tests\Constraint;
+
+use PHPUnit\Framework\Constraint\Constraint;
+
+/**
+ * A constraint that checks a certain source with provided mime type.
+ */
+class ImageHasSource extends Constraint {
+
+	/**
+	 * The requested mime type.
+	 *
+	 * @var string
+	 */
+	protected $mime_type;
+
+	/**
+	 * Determines whether we need to check for absence or for existence of the mime type.
+	 *
+	 * @var bool
+	 */
+	protected $is_not;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string $mime_type The requested mime type.
+	 */
+	public function __construct( $mime_type ) {
+		$this->is_not    = false;
+		$this->mime_type = $mime_type;
+	}
+
+	/**
+	 * Tells to check for absence of the mime type.
+	 */
+	public function isNot() {
+		$this->is_not = true;
+	}
+
+	/**
+	 * Returns a string representation of the constraint.
+	 *
+	 * @return string String representation of the constraint.
+	 */
+	public function toString(): string {
+		return sprintf(
+			'%s the source with the "%s" mime type',
```

Comment:
```suggestion
			'It %s the source with the "%s" mime type',
```

Reply:
Same here. We can't use `It` here because this message is used in the phpunit to compose success/failure messages.